src/Uno.UI.Runtime.Skia.MacOS/UnoNativeMac/UnoNativeMac/UNONative.m

```diff
@@ -115,9 +115,20 @@ bool uno_native_is_attached(NSView<UNONativeElement>* element)
 
 void uno_native_measure(NSView<UNONativeElement>* element, double childWidth, double childHeight, double availableWidth, double availableHeight, double* width, double* height)
 {
-    CGSize size = element.subviews.firstObject.frame.size;
-    *width = size.width;
-    *height = size.height;
+    double resolvedWidth = isfinite(availableWidth) ? availableWidth : (isfinite(childWidth) ? childWidth : 0);
+    double resolvedHeight = isfinite(availableHeight) ? availableHeight : (isfinite(childHeight) ? childHeight : 0);
+
+    if (resolvedWidth <= 0)
+    {
+        resolvedWidth = element.bounds.size.width;
```

Comment:
If `childHeight` is `0` do you want to the element to be `0` or the current size of the `element` ?

If the later you better not initialize to `0` for `resolve*` (e.g. use `-1`) and check for `< 0`.

Reply:
`resolvedWidth = size.width;`